File: Content/Sorting/Python/Bubble.py
def BubbleSort( arr ):
    """ BubbleSort sorts an array using the bubble sort algorithm """

    sortedArr = arr

    for _ in arr:
        for i in range(len(arr)-1):
            if arr[i] > arr[i+1]:
                arr[i], arr[i+1] = arr[i+1], arr[i]

    return sortedArr


if __name__ == "__main__":
    from random import randint
    from time import time

    # running test cases:
    testArr = [ randint(0, 1000) for i in range(50) ]

    print("Unsorted Array: ", testArr); start = time()
    print("Sorted Array: ", BubbleSort(testArr))
    print("Time Taken: ", time() - start)

File: Content/Sorting/Golang/sorting/Bubble.go
package sorting

//BubbleSort sorts an array using the ***bubble sort algorithm***
func BubbleSort(arr *[]int64) {

	for i := 0; i < len(*arr); i++ {
		for j := 0; j < len(*arr)-1; j++ {
			if (*(arr))[j] > (*(arr))[j+1] {
				(*arr)[j], (*arr)[j+1] = (*arr)[j+1], (*arr)[j]
			}
		}
	}
}

/*
BubbleSorted sorts an array using the ***bubble sort algorithm*** and returns the sorted array

! Note: This function doesn't **modify** the array passed into the function
*/
func BubbleSorted(arr []int64) []int64 {
	sortedArr := arr

	for i := 0; i < len(sortedArr); i++ {
		for j := 0; j < len(sortedArr)-1; j++ {
			if sortedArr[j] > sortedArr[j+1] {
				sortedArr[j], sortedArr[j+1] = sortedArr[j+1], sortedArr[j]
			}
		}
	}
	return sortedArr
}
